### PR TITLE
172916 - build permit xml

### PR DESCRIPTION
**Pull request by mayur1501364**
This PR consist of code changes for [PBI 172916](https://dev.azure.com/ukhydro/S-100%20Permit%20Generator/_workitems/edit/172916)

- Code changes added
- Unit Test cases added
- Xsd validation and file added

### CODE REVIEW

File: src/UKHO.S100PermitService.Common/Services/PermitService.cs [hunk 1]
```diff
@@ -17,22 +20,30 @@ namespace UKHO.S100PermitService.Common.Services
...
         private readonly IUserPermitService _userPermitService;
         private readonly IProductKeyService _productKeyService;
+        private readonly IOptions<PermitFileConfiguration> _permitFileConfiguration;
         private readonly IS100Crypt _s100Crypt;
         private readonly IOptions<ProductKeyServiceApiConfiguration> _productKeyServiceApiConfiguration;
 
+        private readonly string _schemaDirectory = Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location)!;
+        private readonly string _issueDate = DateTimeOffset.Now.ToString(DateFormat);
+
+        private Dictionary<string, Permit> _permitDictionary = new Dictionary<string, Permit>();
```
Comment (harshal11869) [inline src/UKHO.S100PermitService.Common/Services/PermitService.cs:37]:
```suggestion
        private Dictionary<string, Permit> _permitDictionary = new();
```

File: src/UKHO.S100PermitService.Common/Services/PermitService.cs [hunk 2]
```diff
@@ -117,26 +138,35 @@ private void CreatePermitXml(DateTimeOffset issueDate, string dataServerIdentifi
...
+                    if(productsList.Any(x => x.Id == products.Id))
+                    {
+                        productsList.FirstOrDefault(x => x.Id == products.Id).DatasetPermit.Add(dataPermit);
+                    }
+                    else
+                    {
+                        products.DatasetPermit = new List<ProductsProductDatasetPermit> { dataPermit };
+                        productsList.Add(products);
+                    }
+                    products = new Products();
```
Comment (harshal11869) [inline src/UKHO.S100PermitService.Common/Services/PermitService.cs:167]:
```suggestion
                    var dataPermit = new ProductsProductDatasetPermit
                    {
                        EditionNumber = byte.Parse(cell.LatestEditionNumber),
                        EncryptedKey = "encryptedkey",
                        Filename = cell.CellCode,
                        Expiry = holding.ExpiryDate
                    };
                    
                    if(productsList.Any(x => x.Id == products.Id))
                    {
                        productsList.FirstOrDefault(x => x.Id == products.Id).DatasetPermit.Add(dataPermit);
                    }
                    else
                    {
                        products.DatasetPermit = new List<ProductsProductDatasetPermit> { dataPermit };
                        productsList.Add(products);
                    }
                    products = new Products();
```

**Review by harshal11869 — CHANGES_REQUESTED** [review]

File: src/UKHO.S100PermitService.Common/Services/PermitService.cs (hunk 1 above)
Comment (mayur1501364) [reply src/UKHO.S100PermitService.Common/Services/PermitService.cs]: Changed.

File: src/UKHO.S100PermitService.Common/Services/PermitService.cs (hunk 2 above)
Comment (mayur1501364) [reply src/UKHO.S100PermitService.Common/Services/PermitService.cs]: Fixed the formatting

File: src/UKHO.S100PermitService.Common/IO/PermitReaderWriter.cs [hunk 3]
```diff
@@ -66,5 +70,18 @@ public void WritePermit(string fileContent)
                 throw;
             }
         }
+
+        private string GetTargetNamespace()
+        {
+            var xsdPath = Path.Combine(_schemaDirectory, "XmlSchema", "Permit_Schema.xsd");
```
[the remark this answers is not in the document]
Comment (mayur1501364) [reply src/UKHO.S100PermitService.Common/IO/PermitReaderWriter.cs]: Constants added

**Review by shirinbt14926 — APPROVED** [review]
LGTM

**Comment by Nadeem14456** [comment]
Changes from this PR will be merged into PR #33. Hence Closing this.

**Pull request closed without merging**